### PR TITLE
implemented geometric_till in gnark and some small optimizations

### LINKED ISSUE #258: Use geometric till in gnark

### PR DESCRIPTION
**Pull request by x-senpai-x**
resolves #258 
Implements `geometricTill` optimization for computing multilinear extension of geometric sequences in gnark circuits.

## Changes
- Replaced O(2^m) EQ polynomial approach with O(m) `geometricTill` algorithm
- Added equivalence and edge case tests

## Performance
Constraint reduction scales with domain size:
- m=10: **42.77x** fewer constraints (2063 → 47)
- m=14: **507.1x** fewer constraints (32963 → 65)

Reduced zkPassport attestation circuit `t_attest` from **5,252,539 → 5,114,167 constraints** (~138k reduction, ~2.6% improvement).

### CODE REVIEW

File: recursive-verifier/app/circuit/public_weight_bench_test.go [hunk 1]
```diff
@@ -0,0 +1,250 @@
...
+}
+
+func (c *OptimizedPublicWeightCircuit) Define(api frontend.API) error {
+	result := geometricTill(api, c.X, c.N, c.FoldingRandomness)
+	api.AssertIsEqual(result, c.Result)
+	return nil
+}
+
+// TestPublicWeightConstraintComparison compares constraint counts of both implementations
+func TestPublicWeightConstraintComparison(t *testing.T) {
```
Comment (ashpect) [inline recursive-verifier/app/circuit/public_weight_bench_test.go:145]: This test compares constraint counts but doesn't verify that both implementations produce identical outputs. For a cryptographic circuit, we need correctness verification.

Please add an equivalence test:
```
func TestGeometricTillCorrectness(t *testing.T) {
... something like this
```

File: recursive-verifier/app/circuit/public_weight_bench_test.go [hunk 2]
```diff
@@ -0,0 +1,250 @@
...
+		}
+
+		naiveConstraints := naiveCCS.GetNbConstraints()
+		optConstraints := optCCS.GetNbConstraints()
+		reduction := float64(naiveConstraints) / float64(optConstraints)
+
+		t.Logf("| %2d | %3d | %10d | %9d | %7.1fx  |", tc.m, tc.n, naiveConstraints, optConstraints, reduction)
+	}
+}
+
```
Comment (ashpect) [inline recursive-verifier/app/circuit/public_weight_bench_test.go:208]: Please add certain edge case tests like :  n = 1, n = 2^k, n = 2^k -1 , n = 2

**Review by ashpect — CHANGES_REQUESTED** [review]
Lgtm, added some comments for tests robustness.

Also please add description about constraints saved switching to geometric_till and more details.

**Review by ashpect — APPROVED** [review]
